Source/JavaScriptCore/runtime/JSStringIterator.cpp

```diff
@@ -59,3 +59,50 @@ JSStringIterator* JSStringIterator::clone(ExecState* exec)
 }
 
 } // namespace JSC
+
+#if PLATFORM(IOS) && !USE(APPLE_INTERNAL_SDK)
+U_STABLE int32_t U_EXPORT2
+ios_specific_unorm_normalize(const UChar *source, int32_t sourceLength,
+                             UNormalizationMode mode, int32_t options,
+                             UChar *resultBuffer, int32_t resultLength,
+                             UErrorCode *status) {
+    if (U_FAILURE(*status)) {
+        return 0;
+    }
+    
+    if ( (source == NULL ? sourceLength != 0 : sourceLength < -1) ||
```

Comment:
IMO some of those checks are not necessary. If I understand it correctly both ```status``` and ```resultBuffer``` are meant to be out parameters so they should not affect the status